### ISSUE TITLE
Landing page copy

### ISSUE THREAD

**Issue by trishalim** [garden-co/classic-jazz#457]
whats jazz trying to sell (pick 1, the rest is just a bonus)

* replace traditional tech stacks
  * this is a big bold claim. why should i steer away from the old reliable proven-and-tested way?
* local-first
* a solution for building collaborative apps
* better UX or better DX?

if thats hard to answer.. think of what does jazz do that makes you think to use it instead of x

and what is x?

questions

* how can i see the data for my app?
* where is my data being stored? aside from the local state
* does mesh exist or is it coming soon? (im quite sure it exists at this point but i have no idea how to begin so it kinda looks like just an idea)
* are CoValues just a collective term for CoMap CoList etc

* what does "generalized" mean

  <img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/8d87b379-1480-47e8-96e9-9823ffc1dee6/84d58107-8849-47b2-a3eb-bfde6c8241fa?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS84ZDg3YjM3OS0xNDgwLTQ3ZTgtOTZlOS05ODIzZmZjMWRlZTYvODRkNTgxMDctODg0OS00N2IyLWEzZWItYmZkZTZjODI0MWZhIiwiaWF0IjoxNzI3NDU2OTEyLCJleHAiOjMzMjk4MDE2OTEyfQ.baWp77MX49-nM7qM66sPKTVeFYq23bXbGNMjWx5Yfe0 " alt="image.png" width="307" height="76" />

ideas

* show react code where a user changes data in a form (with useState, onChange etc) vs the jazz way
* and other traditional way vs jazz way

related products

* [https://www.powersync.com](https://www.powersync.com/) ?
* [https://electric-sql.com/](https://electric-sql.com/) ?

TLDR: i dont really know where the data is going

**Comment by aeplay** [comment]
Hey @trishalim thanks for collecting your thoughts on this. Let's discuss it here in public, might be interesting for folks to follow along (you can also reply with GitHub comments from within Linear if that's more comfortable)

This should probably be broken into several issues, but let's discuss all of them here for now.

First I really appreciate your encouragement to focus on one main message, and all of your suggestions are good.

**I would say the main story is actually "replace traditional tech stacks, for vastly better DX"**

- better UX, good collaboration/multiplayer and local-first are bonuses is my opinion
- an important way to make that "replace everything" hard-sell softer is to focus on greenfield apps for now. I think a good slogan to use somewhere is "build your _next_ app with Jazz"
- the x that Jazz replaces is typically Databases + APIs + Blob Storage + some client-side state handling. Jazz replacing that much is what makes it so powerful but also what makes it an extraordinary claim that requires extraordinary evidence 
  - it also replaces many uses of message queues, but since people encounter these less frequently in normal stacks maybe we can make the diagram & the story simpler by removing them from the "before" stack?

Regarding your questions (they're important!):

> how can i see the data for my app?

We're working on something called the Jazz inspector. Currently you have to set it up manually but then it gives you a graphical representation of all state stored for a specific user. Integrating this more tightly as a debug tool with any app that uses Jazz would go a long way in making it clearer "what Jazz does / where your data is"

> where is my data being stored? aside from the local state

On a sync & persistence server, either Jazz Mesh or one you run yourself with `npx jazz-run sync`. The peer address you specify when setting up a Jazz context points to the server you want to use.

> does mesh exist or is it coming soon?

It exists, is the default sync & persistence server and powers all of the example and real apps we have. I think what would really help make it more real is implementing the developer sign-up flow for a proper API key and having at least a minimal dashboard. We can do actual metering & billing later, but it would be good to have this flow for purely psychological reasons ("it's real & ready")

> are CoValues just a collective term for CoMap CoList etc

Yes, can we make this clearer?

> what does "generalized" mean

In traditional backends, your API contains a lot of app-specific business logic that is part of what makes your app your app. In Jazz, important topics like auth, permissions and state sharing are solved in a generalised (app-independent) way, which means all jazz apps can share the same syncing & persistence server and they can have all business logic in the client (or in lightweight server-side workers, to talk to external APIs)

Can you think of a better way of saying this succinctly?

> ideas

These are both great and would be two perfect illustrations to back up the two messages
- Jazz is like magical useState
- Jazz replaces entire backends

**Comment by trishalim** [comment]
@aeplay 
> "replace traditional tech stacks, for vastly better DX"

Great to have clarity on this. If that's what we want to focus on, then that means "instant sync" no longer fits as a headline, right? "build your next app with Jazz" would be closer to it. It sounds like Jazz gets you faster from your idea to a functional app.

> the x that Jazz replaces is typically Databases + APIs + Blob Storage + some client-side state handling

It would be great to have a section to explain each of those + about the sync

> developer sign-up flow for a proper API key and having at least a minimal dashboard
> for purely psychological reasons

This is a great idea

More questions
- ~What's a backend worker? (I googled it but couldn't find anything)~ Figured it out
- Why are all the examples behind a login screen?

**Comment by aeplay** [comment]
@trishalim 

Thoughts about: "Build your next app with sync"

> Why are all the examples behind a login screen?

To make clear that Jazz also handles auth/user identity (many similar solutions don't) and to give you a quick way to try out multi-user interactions

**Comment by aeplay** [comment]
Also pulling in your comment from the PR:

https://github.com/gardencmp/jazz/pull/449#issuecomment-2379130889

I like your iteration, let's try to come up with something that works with "Build your next app with sync" (if you like that)

Trying something new here:

"Build your next app with sync.

Jazz is an open-source framework for building apps around local data that is synced between devices, users and servers. This replaces most things you usually need a backend for, making it super fast and fun to build and deploy apps."

**Comment by trishalim** [comment]
@aeplay I think that is jumping from one point to another (sync -> replacing backend). I realized this is not that big of a jump considering a lot of the complexity comes from implementing the sync. But it starts off with the sync, so it doesn't align with our "main story" to "replace traditional tech stacks, for vastly better DX"

I tried to improve it anyway

### Build your next app with instant sync
Jazz is the open-source framework for building full-stack apps with real-time data sync across users and devices. We take care of your database, APIs, message queues, UI states, file uploads, and more, so that you can focus on shipping.

This one is more aligned to "replace traditional tech stacks, for vastly better DX"

### Full-stack development made easy
Jazz is the open-source framework for building full-stack apps, without 90% of the complexity. Real-time sync, auth, permissions, instant UI updates, and more. It all works from day one.

I don't know if it's accurate to say full-stack though

### App development made easy
Jazz is the open-source framework for building apps without 90% of the backend and infrastructure complexity. Real-time sync, storage, auth, permissions, instant UI updates, file uploads, and more. It all works from day one.

Or this
<img width="1272" alt="image" src="https://github.com/user-attachments/assets/b87994a4-1e3b-47b1-ac86-9747eb0f7267">

**Comment by aeplay** [comment]
Yeah especially the last one is really getting there for me, will think some more about it.

Thanks for your exploration

**Comment by trishalim** [comment]
To reiterate what was said in the call: [https://www.instantdb.com/](https://www.instantdb.com/) is a "common" alternative, with good DX and storytelling, but lacking in features (such as permissions)

**Comment by trishalim** [comment]
Gathering feedback from [hacker news](https://news.ycombinator.com/item?id=41748912)

These are the doubts:

1. "I'm more than a little worried about the following all being Coming Soon (cursors and carets, 2 way sync, video presence and calls). All of these are the key reasons I would be evaluating this framework to handle my data. All of these are not fully implemented yet."
2. "People will be quick to create apps with this. But what happens if I want to implement something that is not supported by the framework yet?"
3. "It's neat that it's quick to start, but I think that's mostly because the framework has only been evaluated by making tiny toy examples."
4. "Its one thing to use this service on the basis of encryption claims - its another thing to have to clean up the mess from a forgotten API key being leaked somewhere... is there, therefore, a third party involved in an audit of Jazz?"
5. "Plus the sync only works between clients AFAICT — literally all the listed backends are also “coming soon”. Which is troubling. When will backends be supported?"
6. "I worry they’re going to be too rigid for some applications."
7. "Nice, but react."

Comparisons

1. Firebase
2. InstantDB

Other feedback

1. "Where are the information about the company? I cannot find an About Us, where they are based, etc?"
2. Broken email newsletter
3. "your website hijacked the back button, I had to spam click back like 30 times to get back to this hacker news comment thread"

**Comment by trishalim** [comment]
Relevant: 

[Namya @ Supafast on Twitter / X](https://x.com/namyakhann/status/1843600732206854339)

**Comment by trishalim** [comment]
Using other products to explain Jazz could also be helpful (seeing how Firebase and InstantDB keeps coming up). Examples:

1. Instant is a modern Firebase. We make you productive by giving your frontend a real-time database.
2. Supabase is an open source Firebase alternative.
3. iPhone described as "a widescreen iPod with touch controls"

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/5c4de608-07af-46b4-8000-be6279db65cb/faa77762-2e94-4f3a-9dd3-bc3778f2a6c7?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS81YzRkZTYwOC0wN2FmLTQ2YjQtODAwMC1iZTYyNzlkYjY1Y2IvZmFhNzc3NjItMmU5NC00ZjNhLTlkZDMtYmMzNzc4ZjJhNmM3IiwiaWF0IjoxNzI4MzkwMDI1LCJleHAiOjMzMjk4OTUwMDI1fQ.3MFiKHhIH-Ijsz_VQmU2BzhJTWGLvS3TdH1F9d1SxKE " alt="image.png" width="1920" height="1080" />

Or comparison pages like [Ghost vs Substack](https://ghost.org/vs/substack/)

**Comment by trishalim** [comment]
Based on the [structure](https://jazz-homepage-git-landing-page-structure-gcmp.vercel.app/) I came up with, here's a few ways we can proceed (in no particular order):

* Rewrite Jazz Mesh section to mention blob storage. In the features section, replace "data & blob storage" with "File uploads".
* In the docs, add a section that talks about the data structures, First-class files & binary data, Secure permissions, authorship & teams, auto-sub. This allows us to be more flexible in removing/simplifying these in the homepage. Right now, if we do that, we lose information.
* Move auth into features section. Create a section dedicated to "coming soon" features (cursors, sync to db, video). 3 looks great in a grid.
* Move auth into features section so that we have a 3-col grid instead of 4-col. 
* In hero, summarize the most important features in a grid (3-4 items)
* "How does jazz work" section - this one is the most challenging

**Comment by trishalim** [comment]
[https://www.convex.dev/](https://www.convex.dev/)

**Comment by trishalim** [comment]
Example of a feature section with loads of info 

[Supafast Sample Work](https://www.figma.com/proto/MR9TL5Jk74aS1dPTIVfRL7/Supafast-Sample-Work?page-id=0%3A1&type=design&node-id=68235-65069&viewport=16585%2C3430%2C0.89&t=KMbDkYpGMZ9sb8CT-1&scaling=min-zoom)

Here they divide the feature section into categories, and have "additional features" at the end

[https://www.figma.com/proto/MR9TL5Jk74aS1dPTIVfRL7/Supafast-Sample-Work?page-id=0%3A1&type=design&node-id=43300-11495&viewport=16585%2C3430%2C0.89&t=KMbDkYpGMZ9sb8CT-1&scaling=min-zoom](https://www.figma.com/proto/MR9TL5Jk74aS1dPTIVfRL7/Supafast-Sample-Work?page-id=0%3A1&type=design&node-id=43300-11495&viewport=16585%2C3430%2C0.89&t=KMbDkYpGMZ9sb8CT-1&scaling=min-zoom)

**Comment by trishalim** [comment]
<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/bf796aa0-5425-4a34-8f20-ff3fa5db19cf/325d61fc-f729-47d6-962b-e0cea9991f47?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS9iZjc5NmFhMC01NDI1LTRhMzQtOGYyMC1mZjNmYTVkYjE5Y2YvMzI1ZDYxZmMtZjcyOS00N2Q2LTk2MmItZTBjZWE5OTkxZjQ3IiwiaWF0IjoxNzI4NTkxMjQ5LCJleHAiOjMzMjk5MTUxMjQ5fQ.yfICxKcdrl1Xj8s7jrTQbNA1J5Db9ziic80ZVtyv_g0 " alt="image.png" width="1309" />

Experimenting with the hero

**Comment by aeplay** [comment]
Love it @trishalim. In general I'm not a huge fan of center-aligned layouts, please may you try what it looks like left-aligned?
We'll want to put mini-code examples on the right side anyways

**Comment by trishalim** [comment]
@aeplay 
<img width="1344" alt="image" src="https://github.com/user-attachments/assets/6694bb57-baba-4a33-8c67-3959c2d44ada">
What code example were you thinking?

**Comment by aeplay** [comment]
@trishalim Something like

Device A (some abstract frame representing a device)

```tsx
const chat = useCoState(Chat, props.chatID);

return (
  <ChatInput
    onSubmit={text => {
      const msg = Message.create({ text }, { owner: chat._owner })
      chat.push(msg);
    }}
  />
)
```

Device B

```tsx
const chat = useCoState(Chat, props.chatID);

return (
  <ChatBody>
    {chat?.map(msg => msg && <ChatBubble msg={msg} key={msg.id} />)}
  </ChatBody>
)
```

Showing mutation and subscription and implied sync.

Thoughts?

**Comment by trishalim** [comment]
@aeplay  it's not very clear what's going on there<br><br>how about we show some code 

```
const chat = useCoState(Chat, props.chatID);
const text = "Hi! Welcome to the chat room."

chat.push(Message.create(
  { text }, 
  { owner: chat._owner)
));
```

and then a mobile device with the chat app open that shows that message

I still don't like it.

**Comment by aeplay** [comment]
> I still don't like it.

😂

what do you think would be better?

**Comment by trishalim** [comment]
@aeplay thinking but haven't come up with anything yet

**Comment by trishalim** [comment]
@aeplay what we're trying to communicate are

- good DX, easy/simple code
- sync

the example would be a real chat app. no login screen

![hero](https://github.com/user-attachments/assets/ac8b4408-daf7-4669-b4df-828d3d8ba9bc)

**Comment by aeplay** [comment]
This is better, I like what you've left out and that you added the schema but I almost want a step 3 that shows how the UI is subscribed to changes. I know it's a lot, but I think it might be doable.

I'm also thinking... devs should be able to "imagine" the UI and by having lots of code we're speaking to them more directly - i.e. we might not even need a visual of the UI

**Comment by trishalim** [comment]
@aeplay so 3 code snippets and no UI. will try

**Comment by trishalim** [comment]
![hero (2)](https://github.com/user-attachments/assets/698e03d9-69c6-4b99-b341-5d7a07dff145)

**Comment by aeplay** [comment]
Nice! A couple more comments:

- it would be good to have useCoState in step 3
- the explanation for step 3 should be something about how the UI updates on both local mutations and remote changes from other users
- not sure about the handwritten font

**Comment by trishalim** [comment]
For the explanation, how about "Instant UI updates synced across users and devices, no queries needed."

I used the handwritten font to separate explanation, kinda like a whiteboard (it goes with the handwritten arrow as well). It's the first handwritten font I found. We can find a better one. Or we can find another way to differentiate the explanations (through color maybe).

**Comment by aeplay** [comment]
"Instant sync & UI updates across users and devices"<br><br>Thoughts about [https://fonts.google.com/specimen/Kalam?preview.text=Instant%20sync%20%26%20UI%20updates%20across%20users%20and%20devices&classification=Handwriting&sort=relevance](https://fonts.google.com/specimen/Kalam?preview.text=Instant%20sync%20%26%20UI%20updates%20across%20users%20and%20devices&classification=Handwriting&sort=relevance)<br>?

**Comment by trishalim** [comment]
Kalam

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/d963f445-c89f-4592-8307-fa87f692e2ad/53a11020-0b03-43c1-af13-34b3102b41c7?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS9kOTYzZjQ0NS1jODlmLTQ1OTItODMwNy1mYTg3ZjY5MmUyYWQvNTNhMTEwMjAtMGIwMy00M2MxLWFmMTMtMzRiMzEwMmI0MWM3IiwiaWF0IjoxNzI4NjUxNjYwLCJleHAiOjMzMjk5MjExNjYwfQ.p-FNKQ429SVkP84RsPPxC547QVHVWQLTS_bkqFE-LyU " alt="image.png" width="413" height="385" />

Sriracha

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/410ea27c-920b-4b66-88c4-e7aceb19ff73/888d8a6d-2fc2-467e-b219-56ab95bb3155?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS80MTBlYTI3Yy05MjBiLTRiNjYtODhjNC1lN2FjZWIxOWZmNzMvODg4ZDhhNmQtMmZjMi00NjdlLWIyMTktNTZhYjk1YmIzMTU1IiwiaWF0IjoxNzI4NjUxNjYwLCJleHAiOjMzMjk5MjExNjYwfQ.CIFQPCb3LihptrbvMqmMjsIyjJMlGrigtJCcztTrgJM " alt="image.png" width="392" height="384" />

Shantell

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/47ce9682-42c9-47fb-8fd4-76fb2d13ebfd/95f5d7ae-2af8-4c78-914e-78b4d318b659?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS80N2NlOTY4Mi00MmM5LTQ3ZmItOGZkNC03NmZiMmQxM2ViZmQvOTVmNWQ3YWUtMmFmOC00Yzc4LTkxNGUtNzhiNGQzMThiNjU5IiwiaWF0IjoxNzI4NjUxNjYwLCJleHAiOjMzMjk5MjExNjYwfQ.cNZBg2PEBghFjLbyySmhJ-k6mgAkWFu07WFhImQkJPs " alt="image.png" width="390" height="389" />

Feel free to play around with it if you want [https://www.figma.com/design/ouDFjrki1xmX8q6XtCiIVk/hero-design?node-id=0-1&t=9vWL0rm1M66Gjf3l-1](https://www.figma.com/design/ouDFjrki1xmX8q6XtCiIVk/hero-design?node-id=0-1&t=9vWL0rm1M66Gjf3l-1)

**Comment by aeplay** [comment]
Thanks for the menu, I'll take Sriracha please

**Comment by trishalim** [comment]
More ideas from @ndreipoppa 

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/630a37e8-1a80-4608-afd9-95ea8410cbe7/fbc84b9d-42c6-41ad-aed0-3d4cd4b6c7b8?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS82MzBhMzdlOC0xYTgwLTQ2MDgtYWZkOS05NWVhODQxMGNiZTcvZmJjODRiOWQtNDJjNi00MWFkLWFlZDAtM2Q0Y2Q0YjZjN2I4IiwiaWF0IjoxNzI5NTA1NTc2LCJleHAiOjMzMzAwMDY1NTc2fQ.Lbcuub05YJ_B0yQ7PjSyG2SdEpywibhXXSSSgAWZZEU " alt="image.png" width="406" height="397" />

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/0a1b74ae-342a-4e43-8d16-d5070ef1bf98/54b38bb0-d590-436d-bf9a-c66ec07ca001?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS8wYTFiNzRhZS0zNDJhLTRlNDMtOGQxNi1kNTA3MGVmMWJmOTgvNTRiMzhiYjAtZDU5MC00MzZkLWJmOWEtYzY2ZWMwN2NhMDAxIiwiaWF0IjoxNzI5NTA1NTc2LCJleHAiOjMzMzAwMDY1NTc2fQ.EqiaxV322imXW4DPnz2KyVRwWb7e5rMKnbqnzmm3kvU " alt="image.png" width="398" height="380" />

**Comment by aeplay** [comment]
thanks for capturing this here!

**Comment by trishalim** [comment]
Problem vs solution examples

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/4e253f7c-424d-402f-b5ab-a269910e94e1/6cec772c-4dbd-4dd0-ab53-409454b28731?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS80ZTI1M2Y3Yy00MjRkLTQwMmYtYjVhYi1hMjY5OTEwZTk0ZTEvNmNlYzc3MmMtNGRiZC00ZGQwLWFiNTMtNDA5NDU0YjI4NzMxIiwiaWF0IjoxNzI5NTI1OTE4LCJleHAiOjMzMzAwMDg1OTE4fQ.r7OY9oC5H3bkoHQ5YrNgA6e5MRGhdiYu9Twf6Ux0ME8 " alt="image.png" width="1101" height="629" />

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/529251f5-4073-4288-bd72-4bb385d3d271/7e1203e6-1f2e-4811-a480-48df64353e82?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS81MjkyNTFmNS00MDczLTQyODgtYmQ3Mi00YmIzODVkM2QyNzEvN2UxMjAzZTYtMWYyZS00ODExLWE0ODAtNDhkZjY0MzUzZTgyIiwiaWF0IjoxNzI5NTI1OTE4LCJleHAiOjMzMzAwMDg1OTE4fQ.2XeopIB2Hwn-2ibnsFvsK0c3sP9a38VuyRCB_7a4XKs " alt="image.png" width="1256" height="1127" />

<img src="https://uploads.linear.app/acccaaae-2ae4-487e-a83c-7148ce16c659/71316007-32b7-4c29-9fa4-5850da8370df/f3d50a38-647c-4f3f-a3d5-67de45635257?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiL2FjY2NhYWFlLTJhZTQtNDg3ZS1hODNjLTcxNDhjZTE2YzY1OS83MTMxNjAwNy0zMmI3LTRjMjktOWZhNC01ODUwZGE4MzcwZGYvZjNkNTBhMzgtNjQ3Yy00ZjNmLWEzZDUtNjdkZTQ1NjM1MjU3IiwiaWF0IjoxNzI5NTI1OTE4LCJleHAiOjMzMzAwMDg1OTE4fQ.9ryYpiUR7Oews7Xos13TEQlleYQVxI0lNNH8hiY_9TA " alt="image.png" width="1246" height="1123" />

**Comment by trishalim** [comment]
@aeplay Do you agree that our top-three problem is sync? I'd like for us to get specific when talking about the problem.

What makes sync so hard to implement? What can go wrong when you DIY? You've also shown an example of Keynote not being able to resolve conflicts, why does that happen? Why does Notion suck? How long did it take for Linear to finish their sync engine (that's the first thing they worked on right?) so that they could get started on the app?

**Comment by aeplay** [comment]
"yes but" we also just make the DX way better for apps that don't really care about sync (traditional web 2.0 saas apps).<br><br>Do you think we have the luxury of being an obvious cure to "the sync problem" AND being the audacious cure to the "stacks are complex" problem?

**Comment by aeplay** [comment]
We should ask the people from Serious Adopter 5 for some quotes for why it's so hard and what can go wrong. We can also make the believable claim that the sync engine is part of the "moat" of Linear, Figma, etc. without having concrete numbers.

**Comment by aeplay** [comment]
The way to connect the two topics is maybe: traditional stacks are complex and broken.<br><br>This becomes the most apparent when you try to add real-time sync (not made for this kind of throughput) and offline-first / need for conflict resolution (the concept of a single/central source of truth breaks down)

**Comment by trishalim** [comment]
From the [podcast](https://discord.com/channels/1139617727565271160/1270047276223692841/1298618689830125578), you answer who Jazz is for

* not for backend nor fullstack devs who know relational DBs, but for frontend developers who know local state and API requests
* individual developers and small companies who want to build apps quickly, which allows them to target smaller niches because it's very easy to build an app
* bigger companies
  * who don't have the resources to build their own sync engine
  * want to build high quality apps to compete with the best in the market

**Comment by aeplay** [comment]
Thanks for capturing this here, and for listening to the podcast :)<br><br>As discussed, I think the "high-quality" apps is the main universal, but your summary of the finer points is correct

**Comment by trishalim** [comment]
In that case, the headline I came up with here is close https://github.com/gardencmp/jazz/pull/622

**Comment by aeplay** [comment]
(closing this for now, since we're tackling copy in individual issues/PRs now)
